File: backend/fertilar-service/src/main/java/com/fertilar/service/entity/Pila.java
package com.fertilar.service.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Table(name = "pila")
public class Pila {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @Column(length = 255)
    private String ubicacion;

    @Column(name = "fecha_inicio", nullable = false)
    private LocalDate fechaInicio;

    @Column(name = "fecha_fin")
    private LocalDate fechaFin;

    @Column(name = "dias_estimados", nullable = false)
    private Integer diasEstimados = 90;

    @Column(name = "humedad_objetivo", precision = 5, scale = 2, nullable = false)
    private BigDecimal humedadObjetivo = new BigDecimal("43.0");

    @Column(name = "temperatura_objetivo", precision = 5, scale = 2, nullable = false)
    private BigDecimal temperaturaObjetivo = new BigDecimal("30.0");

    @Column(name = "fecha_estimada_fin")
    private LocalDate fechaEstimadaFin;

    @Column(nullable = false, length = 50)
    @Enumerated(EnumType.STRING)
    private Estado estado;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    public enum Estado {
        ACTIVA, FINALIZADA, PAUSADA
    }
}

File: frontend/fertilar-ui/src/types/pila.ts
export type PilaEstado = 'ACTIVA' | 'FINALIZADA' | 'PAUSADA'

export type PilaResumen = {
  id: string
  nombre: string
  estado: PilaEstado
  fechaInicio: string
  ubicacion: string | null
}

export type Pila = PilaResumen & {
  descripcion: string | null
  fechaFin: string | null
  diasEstimados: number
  humedadObjetivo: number
  temperaturaObjetivo: number
  fechaEstimadaFin: string | null
  createdAt: string
}

export type PilaRequest = {
  nombre: string
  descripcion?: string
  ubicacion?: string
  fechaInicio: string
  fechaFin?: string
  diasEstimados?: number
  humedadObjetivo?: number
  temperaturaObjetivo?: number
  estado: PilaEstado
}
